app/components/groups/group-storage.service.js

```diff
@@ -19,21 +19,27 @@
         "panel": panelLoader
       }
 
+      var loadResources = {
+        "groups.list": GroupResource,
+        "trash.list": TrashResource
+      }
+
       function loadData(params){
         var view = sessionStorage.getItem("userView")
-        loaders[view](params)
+        var loadResource = loadResources[$state.current.name]
```

Comment:
You hide actual service calls. It's hard to maintain and confuses other developers (hard to search in the code, etc). Make explicit calls -  like `GroupResource.list()`,  `TrashResource.list()`.
More of this - I think you have to define separate methods in service: one to get items, another to get items in trash